src/ipc-handlers.ts

```diff
@@ -300,91 +313,98 @@ export async function disconnectWpcomSites(
 	event: IpcMainInvokeEvent,
 	list: WpcomSitesToDisconnect
 ) {
-	const userData = await loadUserData();
-	const currentUserId = userData.authToken?.id;
-
-	if ( ! currentUserId ) {
-		throw new Error( 'User not authenticated' );
-	}
+	try {
+		await lockAppdata();
+		const userData = await loadUserData();
+		const currentUserId = userData.authToken?.id;
 
-	const connectedWpcomSites = userData.connectedWpcomSites;
+		if ( ! currentUserId ) {
+			throw new Error( 'User not authenticated' );
+		}
 
-	// Totally unreal case, added it to help TS parse the code below. And if this error happens, we definitely have something wrong.
-	if ( ! Array.isArray( connectedWpcomSites?.[ currentUserId ] ) ) {
-		throw new Error(
-			'Something went wrong, since you are trying to disconnect something, but there are no stored connections yet'
-		);
-	}
+		const connectedWpcomSites = userData.connectedWpcomSites;
 
-	list.forEach( ( { siteIds, localSiteId } ) => {
-		const updatedConnections = connectedWpcomSites[ currentUserId ].filter(
-			( conn ) => ! ( siteIds.includes( conn.id ) && conn.localSiteId === localSiteId )
-		);
+		// Totally unreal case, added it to help TS parse the code below. And if this error happens, we definitely have something wrong.
+		if ( ! Array.isArray( connectedWpcomSites?.[ currentUserId ] ) ) {
+			throw new Error(
+				'Something went wrong, since you are trying to disconnect something, but there are no stored connections yet'
+			);
+		}
 
-		connectedWpcomSites[ currentUserId ] = updatedConnections;
-	} );
+		list.forEach( ( { siteIds, localSiteId } ) => {
+			const updatedConnections = connectedWpcomSites[ currentUserId ].filter(
+				( conn ) => ! ( siteIds.includes( conn.id ) && conn.localSiteId === localSiteId )
+			);
 
-	await saveUserData( userData );
+			connectedWpcomSites[ currentUserId ] = updatedConnections;
+		} );
 
-	return connectedWpcomSites[ currentUserId ].filter( ( conn ) =>
-		list.some( ( { localSiteId } ) => conn.localSiteId === localSiteId )
-	);
+		await saveUserData( userData );
+	} finally {
+		await unlockAppdata();
+	}
 }
 
 export async function updateConnectedWpcomSites(
 	event: IpcMainInvokeEvent,
 	updatedSites: SyncSite[]
 ) {
-	const userData = await loadUserData();
-	const currentUserId = userData.authToken?.id;
+	try {
+		await lockAppdata();
+		const userData = await loadUserData();
+		const currentUserId = userData.authToken?.id;
 
-	if ( ! currentUserId ) {
-		throw new Error( 'User not authenticated' );
-	}
+		if ( ! currentUserId ) {
+			throw new Error( 'User not authenticated' );
+		}
 
-	const connections = userData.connectedWpcomSites?.[ currentUserId ] || [];
+		const connections = userData.connectedWpcomSites?.[ currentUserId ] || [];
 
-	if ( ! connections.length ) {
-		return;
-	}
+		if ( ! connections.length ) {
+			return userData;
```

Comment:
```suggestion
			return;
```
This function should return `void` now.